pretyping/pretyping.ml

```diff
@@ -478,39 +478,138 @@ let mark_obligation_evar sigma k evc =
     Evd.set_obligation_evar sigma (fst (destEvar sigma evc))
   | _ -> sigma
 
-(* [pretype tycon env sigma lvar lmeta cstr] attempts to type [cstr] *)
-(* in environment [env], with existential variables [sigma] and *)
-(* the type constraint tycon *)
+type 'a pretype_fun = ?loc:Loc.t -> program_mode:bool -> poly:bool -> bool -> type_constraint -> GlobEnv.t -> evar_map -> evar_map * 'a
+
+type pretyper = {
```

Comment:
Nit: you can maybe use `ocamlformat` style formatting?
```
{ foo
; bar
}
```

Reply:
Didn't the ocamlformat style we merged for micromega use
~~~
{ foo
; bar }
~~~
?

Reply:
Indeed, actually we should just run the formatter and stop worrying about this O_O

Reply:
AFAIK we never settled for a coding style and I have a passionate hate for this particular way of writing records...

Reply:
Indeed, it would look like haskell ;-)